lib/Twig/Extension/Core.php

```diff
@@ -1337,3 +1337,188 @@ function twig_array_batch($items, $size, $fill = null)
+{
+    static $cache = array();
+
+    // array
+    if (Twig_Template::METHOD_CALL !== $type) {
+        $arrayItem = is_bool($item) || is_float($item) ? (int) $item : $item;
+
+        if ((is_array($object) && (isset($object[$arrayItem]) || array_key_exists($arrayItem, $object)))
+            || ($object instanceof ArrayAccess && isset($object[$arrayItem]))
+        ) {
+            if ($isDefinedTest) {
+                return true;
+            }
+
+            return $object[$arrayItem];
+        }
+
+        if (Twig_Template::ARRAY_CALL === $type || !is_object($object)) {
+            if ($isDefinedTest) {
+                return false;
+            }
+
+            if ($ignoreStrictCheck || !$env->isStrictVariables()) {
+                return;
+            }
+
+            if ($object instanceof ArrayAccess) {
+                $message = sprintf('Key "%s" in object with ArrayAccess of class "%s" does not exist.', $arrayItem, get_class($object));
+            } elseif (is_object($object)) {
+                $message = sprintf('Impossible to access a key "%s" on an object of class "%s" that does not implement ArrayAccess interface.', $item, get_class($object));
+            } elseif (is_array($object)) {
+                if (empty($object)) {
+                    $message = sprintf('Key "%s" does not exist as the array is empty.', $arrayItem);
+                } else {
+                    $message = sprintf('Key "%s" for array with keys "%s" does not exist.', $arrayItem, implode(', ', array_keys($object)));
+                }
+            } elseif (Twig_Template::ARRAY_CALL === $type) {
+                if (null === $object) {
+                    $message = sprintf('Impossible to access a key ("%s") on a null variable.', $item);
+                } else {
+                    $message = sprintf('Impossible to access a key ("%s") on a %s variable ("%s").', $item, gettype($object), $object);
+                }
+            } elseif (null === $object) {
+                $message = sprintf('Impossible to access an attribute ("%s") on a null variable.', $item);
+            } else {
+                $message = sprintf('Impossible to access an attribute ("%s") on a %s variable ("%s").', $item, gettype($object), $object);
+            }
+
+            throw new Twig_Error_Runtime($message, -1, $source);
+        }
+    }
+
+    if (!is_object($object)) {
+        if ($isDefinedTest) {
+            return false;
+        }
+
+        if ($ignoreStrictCheck || !$env->isStrictVariables()) {
+            return;
+        }
+
+        if (null === $object) {
+            $message = sprintf('Impossible to invoke a method ("%s") on a null variable.', $item);
+        } else {
+            $message = sprintf('Impossible to invoke a method ("%s") on a %s variable ("%s").', $item, gettype($object), $object);
+        }
+
+        throw new Twig_Error_Runtime($message, -1, $source);
+    }
+
+    if ($object instanceof Twig_Template) {
```

Comment:
Can you add a test covering it though ? The C extension currently misses this check but passes tests